src/premiumvideodevice/PremiumVideoEffectDriver.ts

```diff
@@ -0,0 +1,67 @@
+import Logger from '../logger/Logger';
+import PremiumVideoEffectConfig from './PremiumVideoEffectConfig';
+
+const CHANNEL_COUNT = 4; // The number of channels in a video frame
+const MAX_PIXEL_VAL = 255; // Max pixel value in unsigned 8 bit int space
+
+/**
+ * [[PremiumVideoEffectDriver]] Mechanism that drives the data transformation
+ * of individual images/frames in order to apply different ML-Based effects.
+ * Current Dummy Effects:
+ *  blue shift
+ *  red shift
+ * Future Effects:
+ *  bg blur
+ *  bg replacement
+ */
+export default class PremiumVideoEffectDriver {
+  constructor(private logger: Logger, private config: PremiumVideoEffectConfig) {
+    this.logger.info('PremiumVideoEffectDriver Created');
+  }
+
+  /**
+   * Apply a transformation of configured effects onto an input image
+   * @param inputImageData
+   * @returns transformed image data to be pasted onto output canvas
+   */
+  async apply(inputImageData: ImageData): Promise<ImageData> {
+    const imageElements = inputImageData.width * inputImageData.height * CHANNEL_COUNT;
+    let rawOutputData: Uint8ClampedArray = new Uint8ClampedArray(imageElements);
```

Comment:
I would do it this way:
`const transformData = (data, idx) => {
      if (this.config.redShiftEnabled && idx % 4 == 0) {
        return MAX_PIXEL_VAL;
      } else if (this.config.blueShiftEnabled && idx % 4 == 2) {
        return = MAX_PIXEL_VAL;
      } else {
        return data;
      }
};
const rawOutputData: Unit8ClampedArray = inputImageData.map((data, idx) => transformData(data, idex));`
`

Reply:
Changed to something similar